packages/storage/src/providers/s3/apis/getProperties.ts

```
@@ -1,5 +1,61 @@
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO: pending implementation
-export declare const getProperties: (params: any) => Promise<any>;
+import { HeadObjectInput, headObject } from '../../../AwsClients/S3';
+import { StorageOptions, StorageOperationRequest } from '../../../types';
+import { assertValidationError } from '../../../errors/utils/assertValidationError';
+import { StorageValidationErrorCode } from '../../../errors/types/validation';
+import { GetPropertiesException, S3GetPropertiesResult } from '../types';
+import { resolveStorageConfig, getKeyWithPrefix } from '../utils';
+import { resolveCredentials } from '../utils/resolveCredentials';
+
+/**
+ * Get Properties of the object
+ *
+ * @param {StorageOperationRequest} The request object
+ * @return {Promise<S3GetPropertiesResult>} Properties of the object
+ * @throws service: {@link GetPropertiesException}
+ * - S3 Service errors thrown while getting properties
+ * @throws validation: {@link StorageValidationErrorCode } - Validation errors thrown
+ *
+ */
+export const getProperties = async function (
+	req: StorageOperationRequest<StorageOptions>
+): Promise<S3GetPropertiesResult> {
+	const { defaultAccessLevel, bucket, region } = resolveStorageConfig();
+	const { identityId, credentials } = await resolveCredentials();
+	const {
+		key,
+		options: { accessLevel },
+	} = req;
+	let targetIdentityId;
+	if (req?.options?.accessLevel === 'protected') {
+		targetIdentityId = req.options?.targetIdentityId;
+	}
+	assertValidationError(!!key, StorageValidationErrorCode.NoKey);
+	const finalKey = getKeyWithPrefix(
+		accessLevel ?? defaultAccessLevel,
+		targetIdentityId ?? identityId,
+		key
+	);
+	const headObjectOptions = {
+		accessLevel,
+		targetIdentityId: identityId,
+		region,
+		credentials,
+	};
+	const headObjectParams: HeadObjectInput = {
+		Bucket: bucket,
+		Key: finalKey,
+	};
+	const response = await headObject(headObjectOptions, headObjectParams);
```

I just realize we need to throw an error with recoverSuggestion here. Suggesting customer needs to create either the bucket or the object because they do not exist

---

we are doing it in while getting it from `resolveStorageConfig`. Do we need to it here again?

---

Thanks for the suggestion. Updated.